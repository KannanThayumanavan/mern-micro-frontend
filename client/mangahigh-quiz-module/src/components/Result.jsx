import React, { Fragment, useEffect, useState } from 'mangahigh-libs/react';
import Button from 'mangahigh-components/Button';
import { calculateResult } from '../utils/utils';
import PropTypes from 'prop-types';

const Result = ({
	questions, updateProgress,
}) => (
	<div className="p-3">
		<h5 className="p-3">You scored {calculateResult(questions)} out of {questions.length}</h5>
		{updateProgress && (
			<Button
			  type='success'
			  buttonLabel='Play again'
			  buttonOnClick={() => updateProgress('playAgain')}
			/>
		)}
	</div>		
);

Result.propTypes = {
	questions: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		question: PropTypes.string.isRequired,
		options: PropTypes.arrayOf(PropTypes.string).isRequired,
		userAnswer: PropTypes.string,
		answer: PropTypes.string.isRequired,
	})),
	updateProgress: PropTypes.func.isRequired,
};
export default Result;




