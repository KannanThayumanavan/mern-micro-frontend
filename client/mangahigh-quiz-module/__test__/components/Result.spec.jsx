import React from 'react';
import { shallow } from 'enzyme';
import Result from '../../src/components/Result';

jest.mock('mangahigh-libs/react', () => jest.requireActual('react'),
	{ virtual: true }
);

jest.mock('mangahigh-components/Button', () => 'Button'			
, { virtual: true });

describe('Test Result Component', () => {
	const mockCallback = jest.fn();
	const props = {
		questions: [
			{
				id: 'test_id',
				question: 'test question 1',
				options: ['test option 1', 'test option 2'],
				userAnswer: 'test option 1',
				answer: 'test option 2',
			}
		],
		updateProgress: mockCallback,
	};
	
	it('should render correctly with required props', () => {
		const renderedModule = shallow(<Result {...props} />);
    expect(renderedModule).toMatchSnapshot();
	});

	it('should callback when Button clicked', () => {
		const renderedModule = shallow(<Result {...props} />);		
    renderedModule.find('Button').props().buttonOnClick(); // see QuizIntro.spec.jsx line#26 
		expect(mockCallback).toHaveBeenCalledWith('playAgain');
	});
});