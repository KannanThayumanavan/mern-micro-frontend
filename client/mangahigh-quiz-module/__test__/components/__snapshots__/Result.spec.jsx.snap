// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Result Component should render correctly with required props 1`] = `
<div
  className="p-3"
>
  <h5
    className="p-3"
  >
    You scored 
    0
     out of 
    1
  </h5>
  <Button
    buttonLabel="Play again"
    buttonOnClick={[Function]}
    type="success"
  />
</div>
`;
